packages/reactivity/__tests__/ref.spec.ts

```diff
@@ -80,6 +80,14 @@ describe('reactivity/ref', () => {
     expect(typeof (c.value.b + 1)).toBe('number')
   })
 
+  it('should pass type check on ref', () => {
+    const arrRef = ref([1, new Map<any, any>(), ref('1')])
+    const [one, two, three] = arrRef.value
+    expect(one).toBe(1)
```

Comment:
This is purely runtime and does not constitute a type check.
To force TS to assert the type for us we need have a statement like `one + 1`, `two.has(1)` and `three + 1` here.